queue_and_stack/dll_queue.py

```diff
@@ -1,19 +1,30 @@
-import sys
-sys.path.append('../doubly_linked_list')
+# import sys
+# sys.path.append('../doubly_linked_list')
 from doubly_linked_list import DoublyLinkedList
 
 
 class Queue:
     def __init__(self):
         self.size = 0
         # Why is our DLL a good choice to store our elements?
-        # self.storage = ?
+
+        # -> Because LLs are O(1) inserting and deleting nodes from head and tail. And that is what a queue is about:
+        # adding values to head and removing from tail
+
+        self.storage = DoublyLinkedList()
 
     def enqueue(self, value):
-        pass
+        self.size += 1
+        self.storage.add_to_head(value)
 
     def dequeue(self):
-        pass
+        if self.size == 0:
+            return
+        else:
+            self.size -= 1
+            value = self.storage.tail.value
+            self.storage.remove_from_tail()
+            return value
 
     def len(self):
-        pass
+        return self.size
```

Comment:
It's nice that you are keeping track of the length so as to have better control on the data structure well done.
This can be bette improved like so
```
def len(self):
        # pass
        # get number of items in stack or queue
        if self.size > 0:
            return self.size
        else:
            return 0
```